### DISCUSSION TITLE
Error saat baca file csv baru (coba oprek data di luar materi)

### DISCUSSION THREAD

**Discussion by imronovsky2000** [teamalgoritma/community#870] [Q&A] [answered]
```
Saat mencoba buat data csv baru (folder :data_sekolah file :sekolah.csv) dengan code berikut :
sekolah = pd.read_csv('data_sekolah/sekolah.csv')
Error:
```
{
	"name": "ParserError",
	"message": "Error tokenizing data. C error: Expected 1 fields in line 81, saw 3
",
	"stack": "---------------------------------------------------------------------------
ParserError                               Traceback (most recent call last)
Cell In[186], line 1
----> 1 sekolah = pd.read_csv('data_input2/sekolah.csv')

File c:\\Users\\User\\miniconda3\\envs\\yoda_da\\lib\\site-packages\\pandas\\io\\parsers\\readers.py:912, in read_csv(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, date_format, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, encoding_errors, dialect, on_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options, dtype_backend)
    899 kwds_defaults = _refine_defaults_read(
    900     dialect,
    901     delimiter,
   (...)
    908     dtype_backend=dtype_backend,
    909 )
    910 kwds.update(kwds_defaults)
--> 912 return _read(filepath_or_buffer, kwds)

File c:\\Users\\User\\miniconda3\\envs\\yoda_da\\lib\\site-packages\\pandas\\io\\parsers\\readers.py:583, in _read(filepath_or_buffer, kwds)
    580     return parser
    582 with parser:
--> 583     return parser.read(nrows)

File c:\\Users\\User\\miniconda3\\envs\\yoda_da\\lib\\site-packages\\pandas\\io\\parsers\\readers.py:1704, in TextFileReader.read(self, nrows)
   1697 nrows = validate_integer(\"nrows\", nrows)
   1698 try:
   1699     # error: \"ParserBase\" has no attribute \"read\"
   1700     (
   1701         index,
   1702         columns,
   1703         col_dict,
-> 1704     ) = self._engine.read(  # type: ignore[attr-defined]
   1705         nrows
   1706     )
   1707 except Exception:
   1708     self.close()

File c:\\Users\\User\\miniconda3\\envs\\yoda_da\\lib\\site-packages\\pandas\\io\\parsers\\c_parser_wrapper.py:234, in CParserWrapper.read(self, nrows)
    232 try:
    233     if self.low_memory:
--> 234         chunks = self._reader.read_low_memory(nrows)
    235         # destructive to chunks
    236         data = _concatenate_chunks(chunks)

File c:\\Users\\User\\miniconda3\\envs\\yoda_da\\lib\\site-packages\\pandas\\_libs\\parsers.pyx:812, in pandas._libs.parsers.TextReader.read_low_memory()

File c:\\Users\\User\\miniconda3\\envs\\yoda_da\\lib\\site-packages\\pandas\\_libs\\parsers.pyx:873, in pandas._libs.parsers.TextReader._read_rows()

File c:\\Users\\User\\miniconda3\\envs\\yoda_da\\lib\\site-packages\\pandas\\_libs\\parsers.pyx:848, in pandas._libs.parsers.TextReader._tokenize_rows()

File c:\\Users\\User\\miniconda3\\envs\\yoda_da\\lib\\site-packages\\pandas\\_libs\\parsers.pyx:859, in pandas._libs.parsers.TextReader._check_tokenize_status()

File c:\\Users\\User\\miniconda3\\envs\\yoda_da\\lib\\site-packages\\pandas\\_libs\\parsers.pyx:2025, in pandas._libs.parsers.raise_parser_error()

ParserError: Error tokenizing data. C error: Expected 1 fields in line 81, saw 3
"
}
```
Mohon bantuannya ya .....

**Comment by ranydwi** [comment, marked as answer]
Dear Pak @imronovsky2000 ,

Terima kasih untuk pertanyaan dan kesediaan waktunya untuk menunggu. 

Untuk error yang Bapak alami sebetulnya solving errornya cukup beragam, mungkin ada problem dengan
- separator/pembatas dalam data Bapak
- baris pertama/row pertama

Boleh Pak discreenshotkan terlebih dahulu bentuk datanya sembari bisa dicoba juga untuk menambahkan parameter `pd.read_csv('file1.csv', sep=**'delimiter'**, on_bad_lines='skip')` -> untuk delimiter bisa diganti dengan pemisah pada file csv Bapak.

Regards,
Rany

**Comment by ramdhinafinita** [comment]
Halo @imronovsky2000 , untuk menjaga agar forum diskusi berjalan kondusif, karena tidak ada respon lanjutan setelah 7 hari dari respon terakhir, kami izin menandai pertanyaan sudah terjawab dan menutup diskusi ini. Semoga dapat dipahami dan terima kasih.
